### PR TITLE
feat(linux): Add `FIREZONE_DNS_CONTROL` env var to choose which DNS control method to use

### PR DESCRIPTION
**Pull request by ReactorScram**
The Docker image for the client is opted in to this new feature. The bare `linux-client-x64` exe is not. I don't know if users are using the Docker images?

I wanted to use CLI args, but the DNS control code ("config" or "control"? Or "SplitDNS"?) has to run at the end of `set_iface_config`, which on Linux runs in a worker, so I couldn't figure out how to move it into `on_set_interface_config` in the callbacks. Maybe there is a way, but the env var results in a small diff.

### CODE REVIEW

File: rust/connlib/tunnel/src/device_channel/tun_linux.rs [hunk 1]
```diff
@@ -231,8 +237,30 @@ async fn set_iface_config(config: InterfaceConfig, _: Vec<IpAddr>, handle: Handl
...
+            }
+            Some(DnsControlMethod::NetworkManager) => {
+                // TODO: Cooperate with NetworkManager
+            }
+            Some(DnsControlMethod::Systemd) => {
+                // TODO: Cooperate with `systemd-resolved`
+                tracing::info!("Will use `systemd-resolved`");
+            }
+        }
+    }
```
Comment (thomaseizinger) [inline rust/connlib/tunnel/src/device_channel/tun_linux.rs]: I am happy to put this here if we agree that this is technical debt that we should fix ASAP once all the `async`, `Arc`, `Mutex` and worker cruft is removed from connlib.

Comment (thomaseizinger) [reply rust/connlib/tunnel/src/device_channel/tun_linux.rs]: How far up the call-chain can we push this before it gets really ugly with interface changes? I think I'd prefer it to be ugly so it serves as a reminder that we need to fix it.

Comment (thomaseizinger) [reply rust/connlib/tunnel/src/device_channel/tun_linux.rs]: Wherever we end up putting it, lets record an issue to capture the tech-debt.

Comment (ReactorScram) [reply rust/connlib/tunnel/src/device_channel/tun_linux.rs]: Agreed. I opened an issue here for debt for both this and the systemd PR https://github.com/firezone/firezone/issues/3636

In the latest commit I moved it up to the `new` function so it's pretty ugly. Technically if the FFI changed we could pass it through there, but it'll have to be removed again if we can get the `on_set_interface_config` callback or any other callback to set up DNS in the client.

**Review by thomaseizinger — APPROVED** [review]
LGTM

Thank you for opening the tech-debt issue.

**Pull request merged**